clang/include/clang/Basic/Attr.td

```diff
@@ -3646,6 +3646,14 @@ def Unavailable : InheritableAttr {
   let MeaningfulToClassTemplateDefinition = 1;
 }
 
+def CanDiscard : InheritableAttr {
+  let Spellings = [CXX11<"clang", "candiscard">,
```

Comment:
We should NOT be doing the `GCC` spelling, this adds a `[[gnu::...]]` spelling, which we absolutely don't want. 

We should be using the `Clang` spelling ONLY for these (so : `let Spellings = [Clang<"candiscard">];`).

This should add a `__attribute__((candiscard))`, plus the two `[[clang::candiscard]]` spellings, which are all the ones we want.